### ISSUE TITLE
Code indexing: add support for `manticoresearch`

### ISSUE THREAD

**Issue by pat-s** [go-gitea/gitea#20889]
### Feature Description

No experiences with this tool yet but it looks promising, both in resource usage and speed (compared to `elasticsearch`).
Also seems to have an HA-ready helm chart.

https://manticoresearch.com/blog/manticore-alternative-to-elasticsearch/

**Comment by wxiaoguang** [comment]
> Drop-in replacement for E in the ELK stack 

So just use it

**Comment by PKizzle** [comment]
@wxiaoguang Did you test manicoresearch with gitea? One of the maintainers of the project said this:

> We do get compared to Elasticsearch a lot. While we support some of its APIs, Manticore isn't a drop-in replacement. 

Source: https://news.ycombinator.com/item?id=44658978

**Comment by TheFox0x7** [comment]
Funny. [Their repository](https://github.com/manticoresoftware/manticoresearch) says otherwise:

> Easy to use open source fast database for search | Good alternative to Elasticsearch now | Drop-in replacement for E in the ELK stack 

They might want to revise it then. Also we can't exactly use the api for this unless we roll out our own client as their official one lacks license https://github.com/manticoresoftware/manticoresearch-go and to quote github on this:
> without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work.

**Comment by wxiaoguang** [comment]
> Did you test manicoresearch with gitea?

No. This issue is inactive for 4 years. So anyone who needs it or has interests on it can try. Otherwise it won't get any progress.

**Comment by wxiaoguang** [comment]
By the way, there are dozens of various new "search engines".

If each time a new engine comes out, then an issue says "support it", then Gitea will become a fat app containing various "client SDKs". I don't think it makes sense.
